crates/common/trie/trie.rs

```diff
@@ -144,6 +142,15 @@ impl Trie {
             .unwrap_or(*EMPTY_TRIE_HASH)
     }
 
+    pub fn commit(&mut self) -> Result<(), TrieError> {
+        if let Some(ref root) = self.root {
+            self.state.commit(root)
+        } else {
+            // nothing to commit
+            Ok(())
+        }
+    }
```

Comment:
```suggestion
        if let Some(ref root) = self.root {
            self.state.commit(root)?;
        }
         Ok(())
    }
```
nit: else branch is not needed

Reply:
Solved in [b7b0646](https://github.com/lambdaclass/ethrex/pull/2174/commits/b7b06463ebf159fc3b88b2230e9a0f533bd0fe0d).